contracts/libraries/Protocol.sol

```diff
@@ -100,15 +104,34 @@ library Protocol {
         return pool.settings.oracle.getTwapPriceX128(pool.settings.twapDuration);
     }
 
-    function getMarginRatio(
+    function getTwapPricesWithDeviationCheck(Protocol.Info storage protocol, uint32 poolId)
+        internal
+        view
+        returns (uint256 realPriceX128, uint256 virtualPriceX128)
+    {
+        realPriceX128 = protocol.getRealTwapPriceX128(poolId);
+        virtualPriceX128 = protocol.getVirtualTwapPriceX128(poolId);
+
+        uint16 maxDeviationBps = protocol.pools[poolId].settings.maxVirtualPriceDeviationRatioBps;
+        if (
+            // if virtual price is too off from real price then screw that, we'll just use real price
+            (int256(realPriceX128) - int256(virtualPriceX128)).absUint() >
+            Math.max(realPriceX128, virtualPriceX128).mulDiv(maxDeviationBps, 1e4)
```

Comment:
Replace "Math.max(realPriceX128, virtualPriceX128)" with realPriceX128

Reply:
Consider the following case:
- maxDeviationBps = 1e4 (100% should mean no bounds)
- realPrice = 2000
- virtualPrice = x

Now the condition renders to |x - 2000|  > 2000 which is equivalent to 0 < x < 4000. However there should be no bound on x.

Reply:
No. I think,100% max deviation means that beyond 100% deviation on upside or downside - the oracle price should be used instead of perp price. So your first statement is correct.

Reply:
I see that's price difference %. Weird but okay.